alt_e2eshark/e2e_testing/backends.py

```diff
@@ -111,7 +95,7 @@ def __init__(self, *, device="local-task", hal_target_backend="llvm-cpu", extra_
     
     def compile(self, module_path: str, *, save_to : str = None) -> str:
         vmfb_path = os.path.join(save_to, "compiled_model.vmfb")
-        arg_string = f"--iree-hal-target-backends={self.hal_target_backend} "
+        arg_string = f"--iree-hal-target-device={self.hal_target_device} "
```

Comment:
Note that this `iree-hal-target-device` flag is quite new, and we aren't even using it in-tree in iree-org/iree for many tests yet.